lib/tapioca/generators/base_generator.rb

```diff
@@ -0,0 +1,110 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "active_support/all"
+
+module Tapioca
+  module Generators
+    class BaseGenerator
+      extend(T::Sig)
+      extend(T::Helpers)
+
+      abstract!
+
+      sig { returns(Config) }
+      attr_reader :config
+
+      sig { params(config: Config).void }
+      def initialize(config)
+        @config = config
+        @bundle = T.let(nil, T.nilable(Gemfile))
+        @compiler = T.let(nil, T.nilable(Compilers::SymbolTableCompiler))
+        @loader = T.let(nil, T.nilable(Loader))
+      end
+
+      sig { returns(Gemfile) }
+      def bundle
+        @bundle ||= Gemfile.new
+      end
+
+      sig { returns(Compilers::SymbolTableCompiler) }
+      def compiler
+        @compiler ||= Compilers::SymbolTableCompiler.new
+      end
+
+      sig { returns(Loader) }
+      def loader
+        @loader ||= Loader.new
+      end
+
+      sig { abstract.void }
+      def generate; end
+
+      sig { abstract.params(error: String).void }
+      def error_handler(error); end
+
+      sig { params(command: String, reason: T.nilable(String), strictness: T.nilable(String)).returns(String) }
+      def rbi_header(command, reason: nil, strictness: nil)
+        statement = <<~HEAD
+          # DO NOT EDIT MANUALLY
+          # This is an autogenerated file for #{reason}.
+          # Please instead update this file by running `#{command}`.
+        HEAD
+
+        sigil = <<~SIGIL if strictness
+          # typed: #{strictness}
+        SIGIL
+
+        if config.file_header
+          [statement, sigil].compact.join("\n").strip.concat("\n\n")
+        elsif sigil
+          sigil.strip.concat("\n\n")
+        else
+          ""
+        end
+      end
+
+      sig { params(constant_name: String).returns(Pathname) }
+      def dsl_rbi_filename(constant_name)
+        config.outpath / "#{constant_name.underscore}.rbi"
+      end
+
+      sig { params(constant_names: T::Array[String]).returns(T::Array[Module]) }
+      def constantize(constant_names)
+        constant_map = constant_names.map do |name|
+          [name, Object.const_get(name)]
+        rescue NameError
+          [name, nil]
+        end.to_h
+
+        unprocessable_constants = constant_map.select { |_, v| v.nil? }
```

Comment:
I find it super surprising that calling a method called `constantize` on a few strings would have the side effect of removing files from my file system.

Let's split this into a second method with an explicit name.